test/test_checkpoint.py

```diff
@@ -0,0 +1,83 @@
+# Copyright 2017 Amazon.com, Inc. or its affiliates. All Rights Reserved.
```

Comment:
Was thinking about it, put them in a new file in the end to leave the door open to additional tests for checkpointing.